### ISSUE TITLE
refactor(studio): narrow local Studio to model training, evaluation, and release

### ISSUE THREAD

**Issue by Teakowa** [OWBastion/ocrkit#6]
## Related work

Depends on:

- `OWBastion/owbastion.com#102` platform-owned OCR annotation pipeline;
- #5 platform dataset snapshot importer.

## Background

The current local Studio combines two different responsibilities:

1. production/R2 screenshot import, candidate generation, human annotation/review, and local review state;
2. OCRKit-owned model-lifecycle work such as ROI preparation, training, checkpoint continuation, evaluation, model comparison, publication, and rollback.

Once the platform becomes the authoritative owner of player feedback and reviewed annotations, keeping a second long-lived annotation source of truth in Studio creates drift and duplicated workflow.

## Goal

Refactor the local OCRKit Studio into a focused **Model Studio** that consumes reviewed platform dataset snapshots and owns only OCRKit model-lifecycle operations.

Do not split Studio into a separate repository unless implementation evidence shows a concrete need. It remains tightly coupled to OCRKit training scripts, Rust ROI tooling, evaluation gates, model artifacts, and release channels.

## Target workflow

```text
select/import platform dataset snapshot
→ inspect materialization/provenance
→ ROI preparation and source-level split
→ configure/start or continue training
→ evaluate candidate checkpoint
→ compare against current/reference model
→ publish through existing immutable release gate
→ rollback by selecting an earlier released manifest/channel target
```

## Keep in OCRKit Studio

Preserve or improve the UI for:

- selecting/importing a finalized reviewed dataset snapshot;
- showing import/materialization validation and missing-source errors;
- source-level train/holdout preparation status;
- Rust ROI/crop preparation where applicable;
- training configuration and checkpoint continuation;
- training/smoke execution status and logs appropriate for a local trusted tool;
- evaluation reports and per-field quality comparison;
- explicit model publication through the existing immutable release path;
- release-channel/rollback visibility where already supported safely.

## Retire or demote

After the platform pipeline and #5 are usable, remove or clearly demote Studio features that would otherwise become a second annotation authority:

- browsing production R2 as the normal way to discover screenshots for labeling;
- local human-review queues as the primary source of reviewed truth;
- independent long-lived correction state that is not tied back to platform-reviewed annotation provenance;
- automatic rule/dataset promotion from local ad-hoc review.

Do not delete useful local/synthetic developer fixture workflows merely because production annotation moved to the platform. Keep a clear distinction between developer fixtures/experiments and authoritative production-derived training datasets.

## Migration

Before removing current local review functionality:

- identify whether any reviewed local batches contain unique private labels that have not been migrated/exported to the private datasets repository;
- preserve reproducibility for already-created model artifacts/checkpoints;
- document how an existing local Studio batch is archived or retained for historical reproduction;
- ensure the new snapshot importer can support the next real training cycle end to end.

Do not silently discard private training data or historical model provenance.

## Architecture constraints

- Keep production inference independent of Studio.
- Keep PaddleOCR/training dependencies outside the production API image.
- Reuse existing training/evaluation/release scripts instead of rebuilding their logic in the frontend/backend.
- Keep source-level train/holdout isolation.
- Preserve immutable model artifact and release-channel rollback semantics.
- Studio remains local/trusted tooling; do not turn this issue into hosting a public training service.

## Acceptance criteria

- [ ] Studio can select/import a finalized platform dataset snapshot through #5 and show validation/provenance clearly.
- [ ] A user can prepare data, start/continue training, evaluate, compare, and explicitly publish a candidate without using the old production-R2 annotation workflow.
- [ ] Platform-reviewed annotations are the authoritative production-derived training truth; Studio no longer maintains a competing primary review queue.
- [ ] Existing synthetic/developer fixtures and regression datasets remain supported and distinct from production-derived training data.
- [ ] Existing reproducible local batches/checkpoints are preserved or intentionally archived with documented provenance before old review functionality is removed.
- [ ] Production OCR API/runtime dependencies are unchanged by the Studio refactor.
- [ ] Existing evaluation gates, immutable artifact publication, verification, and rollback behavior remain intact.
- [ ] Studio/training tests and the relevant OCRKit test/evaluation gates pass.

## Non-goals

- Moving training execution into `owbastion.com`.
- Creating a separate public Studio service or repository by default.
- Replacing OCRKit training scripts with frontend-owned implementations.
- Adding player/admin annotation workflow to OCRKit.
- LLM adjudication; see #4.
